### PR TITLE
feat: added product listing page and components

### PR DESCRIPTION
**Pull request by nerdyspook**
**Added Product Listing Page with following components**
> Filter component
> Sort by component
> Product card component

### CODE REVIEW

File: src/components/Navbar/Navbar.jsx [hunk 1]
```diff
@@ -22,6 +25,14 @@ const Navbar = () => {
...
+    useEffect(() => {
+        if (window.location.pathname !== "/") {
+            setNavItemColor("#2d3748");
+        } else {
+            setNavItemColor("#cbd5e1");
+        }
+    }, [location]);
+
     useEffect(() => {
         window.addEventListener("scroll", scrollHeader);
```
Comment (sharathnair9999) [inline src/components/Navbar/Navbar.jsx]: you can do simply window.scrollTo(0,0). avoid giving events to the document. And for this functionality you can make custom hook and re-use it

**Review by sharathnair9999 — APPROVED** [review]
suggested one change

**Review by prathmeshjagtap — APPROVED** [review]
code is well structured, just a suggestion you can make an array of ratings and categories and loop over it. instead of repeating  code

File: src/components/Navbar/Navbar.jsx (hunk 1 above)
Comment (nerdyspook) [reply src/components/Navbar/Navbar.jsx]: Thanks, I have used hook to change navbar color, but I can't use `window.scrollTo(0,0)` as it doesn't serve the same purpose I want.

**Pull request merged**